### PR TITLE
Add support for Azure blob storage

### PR DESCRIPTION
**Pull request by PrasadG193**
This PR adds support for Azure Blob Storage to Kopia

Signed-off-by: Prasad Ghangal <prasad.ghangal@gmail.com>

Tests:
```
$ go test -v ./repo/blob/azure/ -count=1
=== RUN   TestAzureStorage
--- PASS: TestAzureStorage (30.01s)
=== RUN   TestAzureStorageInvalidBlob
--- PASS: TestAzureStorageInvalidBlob (0.52s)
=== RUN   TestAzureStorageInvalidContainer
--- PASS: TestAzureStorageInvalidContainer (0.25s)
=== RUN   TestAzureStorageInvalidCreds
--- PASS: TestAzureStorageInvalidCreds (0.00s)
PASS
ok      github.com/kopia/kopia/repo/blob/azure  30.782s
```

### CODE REVIEW

File: repo/blob/azure/azure_storage_test.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	if stgErr, ok := err.(azblob.StorageError); ok {
+		if stgErr.ServiceCode() == azblob.ServiceCodeContainerAlreadyExists {
+			return
+		}
+	}
+
+	t.Fatalf("failed to create blob storage container: %v", err)
+}
+
+func TestAzureStorage(t *testing.T) {
```
Comment (jkowalski) [inline repo/blob/azure/azure_storage_test.go]: t.Parallel() perhaps?

Comment (PrasadG193) [reply repo/blob/azure/azure_storage_test.go]: Done!

**Comment by jkowalski** [comment]
@PrasadG193 thank you for this contribution, can we set environment variables on travis so that Azure gets regular test coverage? Please DM me on Slack and I can give you access.

Also, would you be able to take a stab at adding this to UI? Should be as simple as cloning S3-related code.

**Comment by PrasadG193** [comment]
Sure @jkowalski. I'll look into UI changes

**Pull request merged**